mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -1087,40 +1093,74 @@ static LogicalResult populateValidationLogic(ModuleOp &module,
   block->push_back(gpuConvCallOp);
 
   mlir::AllocOp gpuResults;
-  if (operation.getValue() == "conv2d") {
+  if (operation.getValue() == "conv2d")
     gpuResults = outputHostAllocOp;
-  } else if (operation.getValue() == "conv2d_bwd_data") {
+  else if (operation.getValue() == "conv2d_bwd_data")
     gpuResults = inputHostAllocOp;
-    // Reset the input tensor
-    auto inputCpuMemsetOp =
-        builder.create<CallOp>(builder.getUnknownLoc(), mcpuMemset4DFuncOp,
-                               ValueRange{inputMemRefCastOp, inputMemsetValue});
-    block->push_back(inputCpuMemsetOp);
-  } else if (operation.getValue() == "conv2d_bwd_weight") {
+  else if (operation.getValue() == "conv2d_bwd_weight")
     gpuResults = filterHostAllocOp;
-    // Reset the filter tensor
-    auto filterCpuMemsetOp = builder.create<CallOp>(
-        builder.getUnknownLoc(), mcpuMemset4DFuncOp,
-        ValueRange{filterMemRefCastOp, filterMemsetValue});
-    block->push_back(filterCpuMemsetOp);
-  }
 
-  // Emit CPU alloc for CPU convolution
-  auto cpuOutputHostAllocOp =
-      builder.create<AllocOp>(builder.getUnknownLoc(), outputMemRefType);
-  block->push_back(cpuOutputHostAllocOp);
+  mlir::AllocOp cpuFilterHostAllocOp, cpuInputHostAllocOp, cpuOutputHostAllocOp;
+  if (operation.getValue() == "conv2d") {
+    cpuFilterHostAllocOp = filterHostAllocOp;
+    cpuInputHostAllocOp = inputHostAllocOp;
+    // Emit CPU alloc for the output tensor
+    cpuOutputHostAllocOp =
+        builder.create<AllocOp>(builder.getUnknownLoc(), outputMemRefType);
+    block->push_back(cpuOutputHostAllocOp);
+
+    // Emit memref cast
+    auto cpuOutputMemRefCastOp = builder.create<MemRefCastOp>(
+        builder.getUnknownLoc(), cpuOutputHostAllocOp,
+        fourDimUnknownSizeMemRefType);
+    block->push_back(cpuOutputMemRefCastOp);
+
+    // Populate initial values of the output tensor
+    auto cpuOutputMemsetOp = builder.create<CallOp>(
+        builder.getUnknownLoc(), mcpuMemset4DFuncOp,
+        ValueRange{cpuOutputMemRefCastOp, outputMemsetValue});
+    block->push_back(cpuOutputMemsetOp);
 
-  // Emit memref cast
-  auto cpuOutputMemRefCastOp = builder.create<MemRefCastOp>(
-      builder.getUnknownLoc(), cpuOutputHostAllocOp,
-      fourDimUnknownSizeMemRefType);
-  block->push_back(cpuOutputMemRefCastOp);
+  } else if (operation.getValue() == "conv2d_bwd_data") {
```

Comment:
Could you consolidate this long if --- else if --- else if --- routine into a common function here. I figure there's mostly duplicated code among three blocks of code.